File: absoluteValuesSumMinization/absoluteValuesSumMinization.py

import math

def absoluteValuesSumMinimization(a: list) -> int:
    is_even = len(a) % 2 == 0
    target_index = len(a) / 2 - 1  if is_even else math.floor(len(a) / 2) 
    return a[int(target_index)]


print(absoluteValuesSumMinimization([2, 4, 7]));
print(absoluteValuesSumMinimization([2, 4, 7, 6]));
print(absoluteValuesSumMinimization([2, 4, 7, 6, 6]));
print(absoluteValuesSumMinimization([2, 4, 7, 6, 6, 8]));
 

File: absoluteValuesSumMinization/absoluteValuesSumMinization.ts
const absoluteValuesSumMinimization = (a: number[]): number => {
    const isEven = a.length % 2 === 0;
    
    const targetIndex = isEven 
        ? a.length / 2 - 1
        : Math.floor(a.length / 2);

    return a[targetIndex];
}


console.time('absoluteValuesSumMinimization x4');

console.log(absoluteValuesSumMinimization([2, 4, 7]));
console.log(absoluteValuesSumMinimization([2, 4, 7, 6]));
console.log(absoluteValuesSumMinimization([2, 4, 7, 6, 6]));
console.log(absoluteValuesSumMinimization([2, 4, 7, 6, 6, 8]));

console.timeEnd('absoluteValuesSumMinimization x4');
